### ISSUE TITLE
Delegate pass-through

### ISSUE THREAD

**Issue by getaaron** [marzapower/MPNumericTextField#3]
I need to respond to `textFieldDidBeginEditing`, but setting the field's `delegate` breaks the functionality.

One possible solution: add

```
@property (nonatomic, weak) id<UITextFieldDelegate> passthroughDelegate;
```

in `MPNumericTextField.h`.

Then, in `MPNumericTextFieldDelegate.m`, change the method signatures to case to `MPNumericTextField`, and call the passthrough delegate.

For example:

``` objc
- (BOOL)textFieldShouldReturn:(MPNumericTextField *)textField
{
  [textField resignFirstResponder];

  if ([textField.passthroughDelegate respondsToSelector:@selector(textFieldShouldReturn:)]) {
    return [textField.passthroughDelegate textFieldShouldReturn:textField];
  } else {
    return YES;
  }
}
```

All of the other methods would need to be implemented as well, for example:

``` objc
- (void) textFieldDidBeginEditing:(MPNumericTextField *)textField {
  if ([textField.passthroughDelegate respondsToSelector:@selector(textFieldDidBeginEditing:)]) {
    [textField.passthroughDelegate textFieldDidBeginEditing:textField];
  }
}
```

**Comment by marzapower** [comment]
Isn't the info found here (https://github.com/marzapower/MPNumericTextField#delegates) sufficient to fix this issue?

**Comment by getaaron** [comment]
That fixes the issue, but requires pasting a bunch of duplicate code in multiple view controllers. (And with https://github.com/marzapower/MPNumericTextField/commit/0fa8552cd6e6be98d4d5800d2ed1720be76bcf2a, requires passing back `textFieldShouldClear:` everywhere as well.)

**Comment by getaaron** [comment]
(I apologize for not being clear about that in the issue description.)

**Comment by marzapower** [comment]
That's perfectly fine. I understand your point, and was thinking about the same new issue with the new `-textFieldShouldClear:` delegate method called. I'll try to find a solution for both these problems.

**Comment by marzapower** [comment]
Solved using 5dfdecac7a647012b7043740ae3350f10759e9ca
